### PR TITLE
change HTTPError details for a list

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: change HTTPError details for a list

**Special notes for your reviewer**:
This HTTPError details field is not used in dashboard right now

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/upgrade.go [hunk 1]
```diff
@@ -159,18 +159,17 @@ func upgradeClusterNodeDeployments(projectProvider provider.ProjectProvider) end
...
 			if err != nil {
-				updateErrors = append(updateErrors, err)
+				updateErrors = append(updateErrors, err.Error())
 			}
 		}
 
 		if len(updateErrors) > 0 {
-			return nil, errors.NewWithDetails(http.StatusInternalServerError, "failed to update some node deployments",
-				fmt.Sprintf("%+q", updateErrors))
+			return nil, errors.NewWithDetails(http.StatusInternalServerError, "failed to update some node deployments", updateErrors)
```
Comment (maciaszczykm) [inline api/pkg/handler/upgrade.go:172]: +1

File: api/pkg/handler/node_v1.go [hunk 2]
```diff
@@ -213,7 +213,7 @@ func getNodeForClusterLegacy(projectProvider provider.ProjectProvider) endpoint.
 
 func createNodeForClusterLegacy(sshKeyProvider provider.SSHKeyProvider, projectProvider provider.ProjectProvider, dcs map[string]provider.DatacenterMeta) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
-		return nil, k8cerrors.NewWithDetails(http.StatusBadRequest, "Creating Nodes is deprecated. Please create a Node Deployment instead", "If you are calling this API endpoint directrly then use POST \"v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments\" instead")
+		return nil, k8cerrors.NewWithDetails(http.StatusBadRequest, "Creating Nodes is deprecated. Please create a Node Deployment instead", []string{"If you are calling this API endpoint directrly then use POST \"v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments\" instead"})
```
Comment (p0lyn0mial) [inline api/pkg/handler/node_v1.go]: typo `directrly`

File: api/pkg/handler/node_v1_test.go [hunk 3]
```diff
@@ -396,7 +396,7 @@ func TestCreateNodeForClusterIsDeprecated(t *testing.T) {
 		{
 			Name:                               "scenario 1: create a node that match the given spec",
 			Body:                               `{"spec":{"cloud":{"digitalocean":{"size":"s-1vcpu-1gb","backups":false,"ipv6":false,"monitoring":false,"tags":[]}},"operatingSystem":{"ubuntu":{"distUpgradeOnBoot":false}}}}`,
-			ExpectedResponse:                   `{"error":{"code":400,"message":"Creating Nodes is deprecated. Please create a Node Deployment instead","details":"If you are calling this API endpoint directrly then use POST \"v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments\" instead"}}`,
+			ExpectedResponse:                   `{"error":{"code":400,"message":"Creating Nodes is deprecated. Please create a Node Deployment instead","details":["If you are calling this API endpoint directrly then use POST \"v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments\" instead"]}}`,
```
Comment (p0lyn0mial) [inline api/pkg/handler/node_v1_test.go]: here to `directrly`

**Review by p0lyn0mial — COMMENTED** [review]
let's fix the typo and this lgtm, thx!

**Comment by zreigz** [comment]
@maciaszczykm @p0lyn0mial thanks for the review. The typo fixed. PTAL

**Comment by p0lyn0mial** [comment]
/lgtm

waiting for @maciaszczykm before approval

**Pull request merged**